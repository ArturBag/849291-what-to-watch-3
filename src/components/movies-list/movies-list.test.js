import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";

const filmsList = [
  {
    title: `Breaking Bad`,
    genre: `Thriller`,
    issuedDate: 2000,
    preview: `../../markup/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    title: `Pirates`,
    genre: `Drama`,
    issuedDate: 1999,
    preview: `../../markup/img/bohemian-rhapsody.jpg`
  },
  {
    title: `Misfits`,
    genre: `Drama`,
    issuedDate: 1998,
    preview: `../../markup/img/macbeth.jpg`
  },
  {
    title: `Terminal`,
    genre: `Horror`,
    issuedDate: 1992,
    preview: `../../markup/img/aviator.jpg`
  },
  {
    title: `Star Wars`,
    genre: `Documentary`,
    issuedDate: 2002,
    preview: `../../markup/img/we-need-to-talk-about-kevin.jpg`
  },
  {
    title: `Forsage`,
    genre: `Crime`,
    issuedDate: 2001,
    preview: `../../markup/img/what-we-do-in-the-shadows.jpg`
  },
  {
    title: `Goal`,
    genre: `Comedy`,
    issuedDate: 2011,
    preview: `../../markup/img/orlando.jpg`
  },
  {
    title: `New Yorker`,
    genre: `History`,
    issuedDate: 2010,
    preview: `../../markup/img/war-of-the-worlds.jpg`
  }
];

it(`Should render movies list`, () => {

  const tree = renderer
    .create(<MoviesList
      moviesList={filmsList}
      onMovieCardTitleClick={() => { }}
      onMoiveCardMouseOver={() => { }}

    />);

  expect(tree).toMatchSnapshot();


});
