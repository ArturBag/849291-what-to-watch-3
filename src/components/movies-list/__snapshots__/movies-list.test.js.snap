// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render movies list 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Breaking Bad"
        height={175}
        src="../../markup/img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Breaking Bad
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pirates"
        height={175}
        src="../../markup/img/bohemian-rhapsody.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Pirates
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Misfits"
        height={175}
        src="../../markup/img/macbeth.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Misfits
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Terminal"
        height={175}
        src="../../markup/img/aviator.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Terminal
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Star Wars"
        height={175}
        src="../../markup/img/we-need-to-talk-about-kevin.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Star Wars
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Forsage"
        height={175}
        src="../../markup/img/what-we-do-in-the-shadows.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Forsage
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Goal"
        height={175}
        src="../../markup/img/orlando.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Goal
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="New Yorker"
        height={175}
        src="../../markup/img/war-of-the-worlds.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        New Yorker
      </a>
    </h3>
  </article>,
]
`;
